modules/samples/producers/kafka/src/main/java/org/wso2/carbon/sample/kafka/KafkaUtil.java

```diff
@@ -48,16 +53,18 @@ public static String getMessageFilePath(String sampleNumber, String filePath, St
 		if (filePath != null && sampleNumber == null) {
 			resultingFilePath = filePath;
 		} else if (filePath == null && sampleNumber != null) {
-			String urlSplitter[] = url.split("/");
-			resultingFilePath = sampleFilPath.replace("sampleNumber", sampleNumber)+urlSplitter[urlSplitter.length-1]+fileExtension;
+			if(format.equalsIgnoreCase("csv")){
```

Comment:
Thilina,
I think it would be better if you can change the string check to "csv".equalsIgnoreCase(format) so that even if 'format' is null there won't be NPE. Or do a null check for format at top. Can you change that?